src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -475,7 +475,14 @@ object ResourceMatcher extends StrictLogging {
 
     val diskResources = groupedResources.getOrElse(Resource.DISK, Seq.empty)
 
-    val resourcesByType: Map[DiskType, Seq[Protos.Resource]] = diskResources.groupBy { r =>
+    val withoutUnsupportedTypes = diskResources.filterNot { r =>
```

Comment:
I'd be inclined to make this code not fail:

```
/Users/tim/src/m8e/marathon/src/main/scala/mesosphere/marathon/state/Volume.scala

 199 |   def fromMesosType(o: Option[Source.Type]): DiskType =
 200 |     o match {
 201 |       case None => DiskType.Root
 202 |       case Some(Source.Type.PATH) => DiskType.Path
 203 |       case Some(Source.Type.MOUNT) => DiskType.Mount
 204 |       case Some(other) => throw new RuntimeException(s"unknown mesos disk type: $other")
 205 |     }
 206 | }
```

Perhaps introduce `DiskType.Unknown(...)`?

Reply:
The problem is that in many places we use something like 
```
if (diskType == DiskType.Root) {
 // handle the root cases
else {
  // handle all other valid cases
}
```
and introducing `DiskType.Unknown` will lead to many changes which can subtly break something. I'd not risk changing that logic.